Orange/widgets/unsupervised/owsom.py

```diff
@@ -824,17 +825,20 @@ def update_output(self):
                     rows = self.get_member_indices(x, y)
                     indices[rows] = self.selection[x, y]
 
+        new_data = self.data.add_column(ContinuousVariable(name='som-x'), self.coordinates[:, 0], True)
+        new_data = new_data.add_column(ContinuousVariable(name='som-y'), self.coordinates[:, 1], True)
+
```

Comment:
This is one way of doing it, but I think it's missing great opportunity.

The new variables could include `compute_value` that would compute the instance's cells based on its data. That is, given data, it would collect the same attributes as those used in this SOM projection and call `SOM.winner_from_weights(self.cont_x, weights, ssum_weights, self.hexagonal)`. Using `SharedComputeValue` this would be efficient.

With this we gain the possibility of running SOM on some data and then assign cells and coordinates to another, independent data. This can be great for any kind of testing, ensuring the sensibility/stability of SOM. For instance, your modifications will allow seeing SOM in, say Scatter plot, so one will be able to observe whether new data clusters similarly as the one used for training.

However, at this point, the cell assignments for "training" data are known and stored (`self.cells` and `self.member_data`), hence there's no need to recompute them. So this code here can stay as it is, and `compute_value` can be added later.

If you're coming tomorrow, @markotoplak or I (but I'm not yet sure I'll come) can help you with that. It's quite easy, but requires knowing one specific mechanism in Orange. We can also help with some fancy numpy indexing to reconstruct coordinates. :)